ospf6d/ospf6_top.c

```diff
@@ -865,6 +866,23 @@ DEFUN (ospf6_interface_area,
 		return CMD_SUCCESS;
 	}
 
+	/* if more than OSPF6_MAX_IF_ADDRS are configured on this interface
+	 * then don't allow ospfv3 to be configured
+	 */
+	ipv6_count = connected_count_by_family(ifp, AF_INET6);
+	if (oi->ifmtu == OSPF6_DEFAULT_MTU && ipv6_count > OSPF6_MAX_IF_ADDRS) {
+		vty_out(vty,
+			"can not configure OSPFv3 on if %s, must have less than %d interface addresses but has %d addresses\n",
+			ifp->name, OSPF6_MAX_IF_ADDRS, ipv6_count);
+		return CMD_WARNING_CONFIG_FAILED;
```

Comment:
Please never refuse config based on state.  State can always be ordered differently, so any look at state while processing config can introduce a race.  If the addresses are added slowly for whatever reasons, we would accept the config here, but if ospf6d is slow we refuse it.  Also, the user might never see the warning if they configure ospf6d first.